From 741a40ecdd6254d308cd59186adaceed0a6644d0 Mon Sep 17 00:00:00 2001
From: Steve Kondik <shade@chemlab.org>
Date: Sat, 1 Dec 2012 15:22:52 -0700
Subject: [PATCH] display: Add option to disable hardware VSYNC

* Set TARGET_NO_HW_VSYNC to disable use of VSYNC

Change-Id: I01f1c8070bfd94d9e332e1aa806799c0560fa482
---
 libhwcomposer/Android.mk |    4 ++++
 libhwcomposer/hwc.cpp    |    6 ++++++
 2 files changed, 10 insertions(+)

diff --git a/libhwcomposer/Android.mk b/libhwcomposer/Android.mk
index 08347d8..ab0a8f2 100644
--- a/libhwcomposer/Android.mk
+++ b/libhwcomposer/Android.mk
@@ -9,6 +9,10 @@ LOCAL_SHARED_LIBRARIES        := $(common_libs) libEGL liboverlay libgenlock \
                                  libexternal libqdutils libhardware_legacy \
                                  libdl libmemalloc libqservice libGLESv1_CM
 
+ifeq ($(TARGET_NO_HW_VSYNC),true)
+    common_flags += -DNO_HW_VSYNC
+endif
+
 LOCAL_CFLAGS                  := $(common_flags) -DLOG_TAG=\"hwcomposer\"
 LOCAL_ADDITIONAL_DEPENDENCIES := $(common_deps)
 LOCAL_SRC_FILES               := hwc.cpp          \
diff --git a/libhwcomposer/hwc.cpp b/libhwcomposer/hwc.cpp
index 62cf16d..57763c4 100644
--- a/libhwcomposer/hwc.cpp
+++ b/libhwcomposer/hwc.cpp
@@ -73,7 +73,9 @@ static void hwc_registerProcs(struct hwc_composer_device_1* dev,
     // Now that we have the functions needed, kick off
     // the uevent & vsync threads
     init_uevent_thread(ctx);
+#ifndef NO_HW_VSYNC
     init_vsync_thread(ctx);
+#endif
 }
 
 //Helper
@@ -458,7 +460,11 @@ static int hwc_device_open(const struct hw_module_t* module, const char* name,
 
         //Setup HWC methods
         dev->device.common.tag          = HARDWARE_DEVICE_TAG;
+#ifdef NO_HW_VSYNC
+        dev->device.common.version      = 0;
+#else
         dev->device.common.version      = HWC_DEVICE_API_VERSION_1_1;
+#endif
         dev->device.common.module       = const_cast<hw_module_t*>(module);
         dev->device.common.close        = hwc_device_close;
         dev->device.prepare             = hwc_prepare;
-- 
1.7.10.4

